tests/member_chain/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,420 @@
+        : $helper->responseBody($defaultUser)
+]->map();
+
+`;
+
+exports[`logical.php 1`] = `
+<?php
+
+$someLongVariableName = (idx(
+    $prop,
+    ['foo' => 'bar', 'bar' => 'foo', 'foobar' => 'barfoor']
+) || [])->map(function ($edge) { return $edge->node; });
+
+($veryLongVeryLongVeryLong || $e)->map(function ($tickets) {
+    $TicketRecord->createFromSomeLongString();
+});
+
+($veryLongVeryLongVeryLong || $e)->map(function ($tickets) {
+  $TicketRecord->createFromSomeLongString();
+})->filter(function ($obj) { return !!$obj; });
+
+($veryLongVeryLongVeryLong || $anotherVeryLongVeryLongVeryLong || $veryVeryVeryLongError)->map(function ($tickets) {
+    $TicketRecord ->createFromSomeLongString();
+});
+
+($veryLongVeryLongVeryLong || $anotherVeryLongVeryLongVeryLong || $veryVeryVeryLongError)->map(function ($tickets) {
+  $TicketRecord->createFromSomeLongString();
+})->filter(function ($obj) { return !!$obj; });
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<?php
+$someLongVariableName = (
+    idx($prop, ['foo' => 'bar', 'bar' => 'foo', 'foobar' => 'barfoor']) || []
+)->map(function ($edge) {
+    return $edge->node;
+});
+
+($veryLongVeryLongVeryLong || $e)->map(function ($tickets) {
+    $TicketRecord->createFromSomeLongString();
+});
+
+($veryLongVeryLongVeryLong || $e)
+    ->map(function ($tickets) {
+        $TicketRecord->createFromSomeLongString();
+    })
+    ->filter(function ($obj) {
+        return !!$obj;
+    });
+
+(
+    $veryLongVeryLongVeryLong ||
+    $anotherVeryLongVeryLongVeryLong ||
+    $veryVeryVeryLongError
+)->map(function ($tickets) {
+    $TicketRecord->createFromSomeLongString();
+});
+
+(
+    $veryLongVeryLongVeryLong ||
+    $anotherVeryLongVeryLongVeryLong ||
+    $veryVeryVeryLongError
+)
+    ->map(function ($tickets) {
+        $TicketRecord->createFromSomeLongString();
+    })
+    ->filter(function ($obj) {
+        return !!$obj;
+    });
+
+`;
+
+exports[`member_chain.php 1`] = `
+<?php
+
+method()->then(function ($x) { return $x; })
+  ["abc"](function ($x) { return $x; })
+  [$abc](function ($x) { return $x; });
+
+($a->a()->b());
+($a)->a()->b();
+$a->a()->b;
+$a->b->a();
+$a->b()->c()->d();
+$a->b->c->d;
+
+$this->loooooooooooong->lookup = (int) $this->getRequest()->getParam(
+    'some-param'
+);
+
+$aVariable = $theThing
+    ->aLongFunction(1, 2, 3)
+    ->anotherReallyLongFunciton('a', 'b', 'c');
+
+$anotherVariables = $theOtherThing
+    ->aLongFunction(1, 2, 3, 4, 5, 6, 7)
+    ->anotherReallyLongFunciton('a', 'b', 'c');
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<?php
+method()->then(function ($x) {
+    return $x;
+})["abc"](function ($x) {
+    return $x;
+})[$abc](function ($x) {
+    return $x;
+});
+
+($a->a()->b());
+($a)->a()->b();
+$a->a()->b;
+$a->b->a();
+$a
+    ->b()
+    ->c()
+    ->d();
+$a->b->c->d;
+
+$this->loooooooooooong->lookup = (int) $this->getRequest()->getParam(
+    'some-param'
+);
```

Comment:
WIP

Reply:
Prettier have difference logic for `this` and regural `variable`. Should we do this as `prettier` or no?

Reply:
Hmm, my first intuition is that in PHP `$this` is not as special as `this` in JS. Do you have a concrete case that would call for different logic?